query/src/sql/planner/binder/aggregate.rs

```diff
@@ -12,114 +12,471 @@
+            }
+            .into()),
+            Scalar::OrExpr(scalar) => Ok(OrExpr {
+                left: Box::new(self.visit(&scalar.left)?),
+                right: Box::new(self.visit(&scalar.right)?),
+            }
+            .into()),
+            Scalar::ComparisonExpr(scalar) => Ok(ComparisonExpr {
+                op: scalar.op.clone(),
+                left: Box::new(self.visit(&scalar.left)?),
+                right: Box::new(self.visit(&scalar.right)?),
+            }
+            .into()),
+            Scalar::FunctionCall(func) => {
+                let new_args = func
+                    .arguments
+                    .iter()
+                    .map(|arg| self.visit(arg))
+                    .collect::<Result<Vec<_>>>()?;
+                Ok(FunctionCall {
+                    arguments: new_args,
+                    func_name: func.func_name.clone(),
+                    // We won't modify data type of any argument during visiting, so it's
+                    // fine to reuse the types.
+                    arg_types: func.arg_types.clone(),
+                    return_type: func.return_type.clone(),
+                }
+                .into())
+            }
+            Scalar::Cast(cast) => Ok(CastExpr {
+                argument: Box::new(self.visit(&cast.argument)?),
+                from_type: cast.from_type.clone(),
+                target_type: cast.target_type.clone(),
+            }
+            .into()),
+
+            // TODO(leiysky): should we recursively process subquery here?
+            Scalar::SubqueryExpr(_) => Ok(scalar.clone()),
+
+            Scalar::AggregateFunction(agg_func) => self.replace_aggregate_function(agg_func),
+        }
+    }
+
+    /// Replace the arguments of aggregate function with a BoundColumnRef, and
+    /// add the replaced aggregate function and the arguments into `AggregateInfo`.
+    fn replace_aggregate_function(&mut self, aggregate: &AggregateFunction) -> Result<Scalar> {
+        let agg_info = &mut self.bind_context.aggregate_info;
+        let mut replaced_args: Vec<Scalar> = Vec::with_capacity(aggregate.args.len());
+
+        for (i, arg) in aggregate.args.iter().enumerate() {
+            let name = format!("{}_arg_{}", &aggregate.func_name, i);
+            let index = self
+                .metadata
+                .add_column(name.clone(), arg.data_type(), None);
 
-    /// Available aliases can be used as group item
-    pub available_aliases: Vec<ColumnBinding>,
+            // Generate a ColumnBinding for each argument of aggregates
+            let column_binding = ColumnBinding {
+                table_name: None,
+
+                // TODO(leiysky): use a more reasonable name, since aggregate arguments
+                // can not be referenced, the name is only for debug
+                column_name: name,
+                index,
+                data_type: arg.data_type(),
+                scalar: Some(Box::new(arg.clone())),
+            };
```

Comment:
We can directly use `create_column_binding`

Reply:
Fixed.